app/models/computacenter/api/cap_usage_update.rb

```diff
@@ -0,0 +1,45 @@
+class Computacenter::API::CapUsageUpdate
+  attr_accessor :cap_type, :ship_to, :cap_amount, :cap_used, :status, :error
+
+  # computacenter cap type => SchoolDeviceAllocation.device_type
+  CAP_TYPES_MAP = {
+    'DfE_RemainThresholdQty|Std_Device' => 'std_device',
+    'DfE_RemainThresholdQty|Coms_Device' => 'coms_device',
+  }.freeze
+
+  # we'll get hashes with string keys from given XML
+  # e.g. {"capType"=>"DfE_RemainThresholdQty|Std_Device", "shipTo"=>"81060874", "capAmount"=>"100", "usedCap"=>"20"}
+  def initialize(string_keyed_hash = {})
+    @cap_type = string_keyed_hash['capType']
+    @ship_to = string_keyed_hash['shipTo']
+    @cap_amount = string_keyed_hash['capAmount']
+    @cap_used = string_keyed_hash['usedCap']
+    @status = 'received'
+    @error = nil
+  end
+
+  def apply!
+    school = School.find_by_computacenter_reference!(ship_to)
+    allocation = school.device_allocations.find_by_device_type!(CAP_TYPES_MAP[cap_type])
+    log_cap_mismatch(allocation) if cap_amount != allocation.allocation
+    allocation.update!(devices_ordered: cap_used)
+    @status = 'succeeded'
+  end
+
+  def log_cap_mismatch(allocation)
+    Rails.logger.warn("CapUsage mismatch: given capAmount: #{cap_amount}, SchoolDeviceAllocation: #{allocation.inspect}")
```

Comment:
I think it'd be helpful to explicitly log the school URN in this message. I know you can look it up from the allocation but it'd save an extra 2 steps...

Reply:
sure, done 👍